pkg/controllers/reconciler.go

```diff
@@ -159,6 +167,72 @@ func (r *reconciler) createObjects(ctx context.Context,
 	return result, nil
 }
 
+func imageGetVersion(image string) string {
+	version := ""
+
+	parts := strings.SplitN(filepath.Base(image), ":", 2)
+
+	if len(parts) == 2 {
+		version = parts[1]
+	}
+
+	return version
+}
+
+func imageGetName(image string) string {
+	name := ""
+
+	parts := strings.SplitN(filepath.Base(image), ":", 2)
+
+	if len(parts) == 2 {
+		name = parts[0]
+	}
+
+	return name
+}
+
+func upgradeToImage(image string) string {
+	out := ""
+
+	ver, e1 := version.ParseSemantic(imageGetVersion(image))
+	minVer, e2 := version.ParseSemantic(ImageMinVersion)
+
+	if e1 == nil && e2 == nil && ver.LessThan(minVer) {
+		out = "intel/" + imageGetName(image) + ":" + ImageMinVersion
+	}
+
+	return out
+}
+
+func UpgradeImages(image *string, initimage *string) (upgrade bool) {
+	if image != nil {
+		if newImage := upgradeToImage(*image); len(newImage) > 0 {
+			*image = newImage
+			upgrade = true
+		}
+	}
+
+	if initimage != nil {
+		if newInitimage := upgradeToImage(*initimage); len(newInitimage) > 0 {
+			*initimage = newInitimage
+			upgrade = true
+		}
+	}
+
```

Comment:
nit: would it make sense to convert this function into a loop with 2 iterations and get rid of upgradeToImage function?